### PR TITLE
Added info, warning, and callout

### PR DESCRIPTION
**Pull request by theannachen**
Added built in WYSIWYG components for info, callout, and warning tags. 

The MDX parsing is also updated to parse those into the MDX correctly.

### CODE REVIEW

File: src/plugins/auto-content/server/services/mdx-service.js [hunk 1]
```diff
@@ -82,6 +82,61 @@ turndownService.addRule("styles", {
...
+    const title = node.getAttribute('title');
+    var titleContent;
+    try{
+      titleContent = node.querySelector('h3').textContent.trim();
+    }
+    catch(e){
+      titleContent = ""
+    }
+
+    const paragraphs = Array.from(node.querySelectorAll('p'));
```
Comment (langdonholmes) [inline src/plugins/auto-content/server/services/mdx-service.js:104]: I think we should avoid using the `<h3>` element as the selector and just operate on the `content` of the `title` directly.

File: src/plugins/ckeditor/admin/src/components/CKEditorInput/plugins/Callout/CalloutEditing.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            // allowed in places where other blocks are allowed (e.g. directly in the root).
+            inheritAllFrom: '$blockObject',
+            allowChildren: '$text',
+        });
+
+        schema.register( 'CalloutContent', {
+            // Cannot be split or left by the caret.
+            isLimit: true,
+            allowIn: 'Callout',
+            allowContentOf: '$root'
```
Comment (langdonholmes) [inline src/plugins/ckeditor/admin/src/components/CKEditorInput/plugins/Callout/CalloutEditing.js:34]: I think we need to change this to disallow `Callout` from being created within `CalloutContent` (we should not allow nesting).

**Review by langdonholmes — CHANGES_REQUESTED** [review]
Hi Anna, this is a great start. Main points are the following:

- Disallow nested elements
- Control the display of the elements using CSS rather than HTML tags. Your CSS should be defined in terms of `.section class="info"` or whatever (sorry I don't know CSS). It can apply the same style as `<h3>` if that's what we want, but the <h3> tag does not (should not) be injected.

**Comment by theannachen** [comment]
> * Control the display of the elements using CSS rather than HTML tags. Your CSS should be defined in terms of `.section class="info"` or whatever (sorry I don't know CSS). It can apply the same style as `<h3>` if that's what we want, but the ###  tag does not (should not) be injected.

Just a question about this point. Right now, all the CSS is in terms of classes. The . in front of the css is indicative of the class name. For example, 

```
.Callout{
<css stuff here>
}
``` 
is referring to anything with the class "Callout," not the tag. Are you suggesting to do CSS in terms of both the tags and the classes (i.e. "for all sections that have the class "Callout")?

Also the for the h3, there is currently no css associated with the header, just the default h3. Do you want me to add something to it?

**Comment by langdonholmes** [comment]
I don't know any css, but I think you are right.

I just want to avoid having auto-generated HTML inside the custom iTELL elements. If we want header formatting, we can define that through CSS instead of injecting an <h#> tag inside the element.

This is partially because CKEditor was deleting the <h#> tag when I moved an element block via drag and drop. So, just to avoid this bug.

Another reason is that we would need to remove the HTML tag when converting to MDX. Seems simpler to just not inject it in the first place.

**Comment by theannachen** [comment]
I think that's related to the listener for the event of the dropping. I will look into alternative options, which will probably actually fix the other issue of not being able to edit the title afterwards as well. I will make new commits with the changes for more review. Thanks!

**Pull request merged**